dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -330,7 +330,7 @@ private void logToolRefresh(final String dashboardPrefix, final Tool tool) {
                         final String manifestDigest = i.getDigest();
                         checksums.add(new Checksum(manifestDigest.split(":")[0], manifestDigest.split(":")[1]));
                     }
-                    Set<Image> images = Collections.singleton(new Image(checksums, repo, tag.getName(), r.getImageID()));
+                    Set<Image> images = Collections.singleton(new Image(checksums, repo, tag.getName(), r.getImageID(), tool.getRegistry()));
                     tag.getImages().addAll(images);
```

Comment:
You don't need to create an intermediate Set; you can just add the image directly by calling `tag.getImages().add(<image>)`.